product_secondary_unit/models/product_secondary_unit_mixin.py

```diff
@@ -0,0 +1,133 @@
+
+    @api.model
+    def _get_secondary_uom_qty_depends(self):
+        if not self._secondary_unit_fields:
+            return []
+        return [self._secondary_unit_fields["qty_field"]]
+
+    @api.depends(lambda x: x._get_secondary_uom_qty_depends())
+    def _compute_secondary_uom_qty(self):
+        for line in self:
+            if not line.secondary_uom_id:
+                line.secondary_uom_qty = 0.0
+                continue
+            elif line.secondary_uom_id.dependency_type == "independent":
+                continue
+            factor = line._get_factor_line()
+            qty_line = line._get_quantity_from_line()
+            qty = float_round(
+                qty_line / (factor or 1.0),
+                precision_rounding=line.secondary_uom_id.uom_id.rounding,
+            )
+            line.secondary_uom_qty = qty
+
+    def _get_default_value_for_qty_field(self):
+        return self.default_get([self._secondary_unit_fields["qty_field"]]).get(
+            self._secondary_unit_fields["qty_field"]
+        )
+
+    def _compute_helper_target_field_qty(self):
+        """Set the target qty field defined in model"""
+        default_qty_field_value = self._get_default_value_for_qty_field()
+        for rec in self:
+            if not rec.secondary_uom_id:
+                rec[rec._secondary_unit_fields["qty_field"]] = (
+                    rec._origin[rec._secondary_unit_fields["qty_field"]]
+                    or default_qty_field_value
+                )
+                continue
+            if rec.secondary_uom_id.dependency_type == "independent":
+                if rec[rec._secondary_unit_fields["qty_field"]] == 0.0:
+                    rec[
+                        rec._secondary_unit_fields["qty_field"]
+                    ] = default_qty_field_value
+                continue
+            # To avoid recompute secondary_uom_qty field when
+            # secondary_uom_id changes.
+            rec.env.remove_to_compute(
+                field=rec._fields["secondary_uom_qty"], records=rec
+            )
+            factor = rec._get_factor_line()
+            qty = float_round(
+                rec.secondary_uom_qty * factor,
+                precision_rounding=rec._get_uom_line().rounding,
+            )
+            rec[rec._secondary_unit_fields["qty_field"]] = qty
+
+    def _onchange_helper_product_uom_for_secondary(self):
+        """Helper method to be called from onchange method of uom field in
+        target model.
+        """
+        if not self.secondary_uom_id:
+            self.secondary_uom_qty = 0.0
+            return
+        elif self.secondary_uom_id.dependency_type == "independent":
+            return
+        factor = self._get_factor_line()
+        line_qty = self._get_quantity_from_line()
+        qty = float_round(
+            line_qty / (factor or 1.0),
+            precision_rounding=self.secondary_uom_id.uom_id.rounding,
+        )
+        self.secondary_uom_qty = qty
```

Comment:
Here, you can add the default_get method.

    @api.model
```
    def default_get(self, fields_list):
        defaults = super().default_get(fields_list)
        if self.secondary_uom_id and not self.env.context.get(
            "skip_default_secondary_uom_qty", False
        ):
            defaults["secondary_uom_qty"] = 1.0
        return defaults

```

Reply:
Ok, PR #1336 incorporated

Reply:
After testing, I still have errors with module **sale_order_secondary_unit** that depends on this module.

`odoo/fields.py:802: UserWarning: Field sale.order.line.product_uom_qty cannot be precomputed as it depends on non-precomputed field sale.order.line.secondary_uom_qty`

Reply:
Is that the only warning you get?